test/web_test.rb

```diff
@@ -18,6 +18,39 @@ def add_pair(job)
   end
 end
 
+# Common test scenarios for both metrics index and detail page.
+module TestScenariosForMetrics
+  extend Minitest::Spec::DSL
+
+  def self.included(base)
+    context "when period param is not provided" do
+      let(:period) { nil }
+      let(:expected_minutes) { 60 }
```

Comment:
This abstract style of testing is not the Sidekiq test suite style at all. Your tests should be directly readable. No modules, no spec DSLs, just:

```ruby
def test_something
  setup_code
  run the code to test
  assert results
end
```

Reply:
> This abstract style of testing is not the Sidekiq test suite style at all. Your tests should be directly readable. No modules, no spec DSLs, just:
> 
> ```ruby
> def test_something
>   setup_code
>   run the code to test
>   assert results
> end
> ```

Completely understood. I've updated the tests to be more in line with the rest.

Reply:
Wow, that looks a lot better, much easier to read. Thank you for the polish.